Userland/Libraries/LibWeb/Painting/ButtonPaintable.cpp

```diff
@@ -43,17 +43,29 @@ void ButtonPaintable::paint(PaintContext& context, PaintPhase phase) const
 
     auto const& dom_node = layout_box().dom_node();
     if (is<HTML::HTMLInputElement>(dom_node) && phase == PaintPhase::Foreground) {
-        auto text_rect = context.enclosing_device_rect(absolute_rect());
+        auto button_rect = context.enclosing_device_rect(absolute_rect());
+        auto text_rect = button_rect;
+
+        // Apply CSS text-indent property to text rect
+        auto text_indent = computed_values().text_indent().to_px(layout_box(), CSSPixels());
+        text_rect.translate_by(text_indent.to_int(), 0);
```

Comment:
This is subtly wrong, because the `text_indent` is in `CSSPixels`, and the `text_rect` is in `DevicePixels`. A CSS pixel is `1px`, but a device pixel may be more or less than that, depending primarily on the display density (a HiDPI screen will have multiple device pixels for each CSS pixel) but also on things like the zoom level.

The simplest option is probably to convert `text_indent` to device pixels here:
```suggestion
        text_rect.translate_by(context.rounded_device_pixels(text_indent), 0);
```